source/common/network/address_impl.h

```diff
@@ -10,6 +10,9 @@
 #include <string>
 
 #include "envoy/network/address.h"
+#include "envoy/network/io_handle.h"
+
+#include "common/network/io_socket_handle_impl.h"
```

Comment:
I don't think this include should be needed. Please check the includes to make sure the the impl header is only includd when an impl is created.

Reply:
It apparently does need to be there- see the discussion below with @alyssawilk. Without this include, listener_impl.cc doesn't compile, even if I add io_socket_handle_impl.h to the hdrs section for listener_lib in the BUILD file. Suggestions?

Reply:
I think you need to include io_socket_handle_impl.h in address_impl.cc as you instantiate an instance there, but here you only need io_handle.h because you don't need anything specific to IoSocketHandle.

You will need to specify the dependency both here and in BUILD file.